Eggstravaganza/Assets/Scripts/ItemsSpawn.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+using Random = UnityEngine.Random;
+public class ItemsSpawn: MonoBehaviour
+{
+    GameObject[] m_Prefabs;
+    [SerializeField]
+    int numItemsToSpawn = 5;
+    [SerializeField]
+    float spawnAreaScale = 1;
+    
+    void Awake()
+    {
+        m_Prefabs = Resources.LoadAll<GameObject>("SpawnItems");
+    }
+    // Function to be called by Game Manager
+    public void SpawnItem()
+    {
+        for (var i = 0; i < numItemsToSpawn; i++)
+        {
+            var pickRand = Random.Range(0, m_Prefabs.Length);
+            var spawnPos = Random.insideUnitSphere * spawnAreaScale;
+            Instantiate(m_Prefabs[pickRand],  new Vector3(spawnPos.x, Random.Range(2, 6), spawnPos.y), Quaternion.identity);
```

Comment:
maybe expose the 2, 6 as like a min/max height spawn parameter in the script as well?